### PR TITLE
Defect/SG-977 - Org domain event logs missing details

### PR DESCRIPTION
**Pull request by JaredSnider-Bitwarden**
## Type of change

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

- For domain verification event logs generated from the background job, create custom system user type of DomainVerification so it can be displayed in the member column of the event logs screen on the client.  
- Create a new device type of server so the client doesn't have to show unknown, and apply this to existing SCIM logs. 
- Fix missing EventSystemUser.SCIM property on user restore / revoke 

## Code changes

* **bitwarden_license/src/Scim/Controllers/v2/UsersController.cs:** Fix missing EventSystemUser.SCIM property on user restore / revoke 

* **src/Core/Enums/DeviceType.cs:** Add new Server device type. 
* **src/Core/Enums/EventSystemUser.cs, src/Core/Services/Implementations/OrganizationDomainService.cs, test/Core.Test/Services/OrganizationDomainServiceTests.cs:** Refactor EventSystemUser SSO property to be DomainVerification per product
* **src/Core/Services/Implementations/EventService.cs:** (1) For existing logs that can have SystemEventUser.SCIM passed in, if it is passed in, then also set device type to server (2) For background job domain verification job logs, set device type to server

Client side results (pending PR and merge):
Domain Verification 
image

SCIM
image

image

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Comment by JaredSnider-Bitwarden** [comment]
@r-tome , I'm new to this style of backend unit tests. I'm not sure where the parameters are coming from for the failing test `LogOrganizationUserEvent_WithEventSystemUser_LogsRequiredInfo`?   I think changing the deviceType in the `EventService.CreateLogOrganizationUserEventsAsync` method maybe is what has broken the test?

File: src/Core/Services/Implementations/EventService.cs [hunk 1]
```diff
@@ -185,6 +185,14 @@ private async Task CreateLogGroupEventAsync(Group group, EventType type, EventSy
             Date = date.GetValueOrDefault(DateTime.UtcNow),
             SystemUser = systemUser
         };
+
+        if (systemUser is EventSystemUser.SCIM)
```
Comment (r-tome) [inline src/Core/Services/Implementations/EventService.cs:189]: All Events with a defined `SystemUser` should have the new `DeviceType.Server` value so I'd move that logic up to the `EventMessage` constructor:

```
var e = new EventMessage(_currentContext)
        {
            OrganizationId = group.OrganizationId,
            GroupId = group.Id,
            Type = type,
            ActingUserId = _currentContext?.UserId,
            ProviderId = await GetProviderIdAsync(@group.OrganizationId),
            Date = date.GetValueOrDefault(DateTime.UtcNow),
            SystemUser = systemUser,
            DeviceType = systemUser.HasValue ? DeviceType.Server : null
        };
```

File: src/Core/Services/Implementations/EventService.cs [hunk 2]
```diff
@@ -249,7 +257,16 @@ private async Task CreateLogOrganizationUserEventsAsync(IEnumerable<(Organizatio
                 ActingUserId = _currentContext?.UserId,
                 Date = date.GetValueOrDefault(DateTime.UtcNow),
                 SystemUser = systemUser
-            });
+            };
+
+            if (systemUser is EventSystemUser.SCIM)
```
Comment (r-tome) [inline src/Core/Services/Implementations/EventService.cs:262]: Same thing as on the `CreateLogGroupEventAsync` method - set `DeviceType = systemUser.HasValue ? DeviceType.Server : null` in the `EventMessage` class instantiation.

**Review by r-tome — CHANGES_REQUESTED** [review]
@JaredSnider-Bitwarden the values for the parameters in the unit tests come from the `BitAutoData` attribute, they are random values so we don't have to define them every time.

You can fix the test by removing the `DeviceType` parameter because in that test scenario since we're writing an Event with a `EventSystemUser` value then we know that the `DeviceType` will be Server. Then change the expected value to this:

```
var expected = new List<IEvent>() {
            new EventMessage()
            {
                IpAddress = ipAddress,
                DeviceType = DeviceType.Server,
                OrganizationId = orgUser.OrganizationId,
                UserId = orgUser.UserId,
                OrganizationUserId = orgUser.Id,
                ProviderId = providerId,
                Type = eventType,
                ActingUserId = actingUserId,
                Date = date,
                SystemUser = eventSystemUser,
            }
        };
```

**Comment by JaredSnider-Bitwarden** [comment]
> @JaredSnider-Bitwarden the values for the parameters in the unit tests come from the `BitAutoData` attribute, they are random values so we don't have to define them every time.
> 
> You can fix the test by removing the `DeviceType` parameter because in that test scenario since we're writing an Event with a `EventSystemUser` value then we know that the `DeviceType` will be Server. Then change the expected value to this:
> 
> ```
> var expected = new List<IEvent>() {
>             new EventMessage()
>             {
>                 IpAddress = ipAddress,
>                 DeviceType = DeviceType.Server,
>                 OrganizationId = orgUser.OrganizationId,
>                 UserId = orgUser.UserId,
>                 OrganizationUserId = orgUser.Id,
>                 ProviderId = providerId,
>                 Type = eventType,
>                 ActingUserId = actingUserId,
>                 Date = date,
>                 SystemUser = eventSystemUser,
>             }
>         };
> ```

Ah, I almost got that right with trying to set the expected deviceType, but I couldn't figure out the AutoData piece. Thank you very much for the explanation.

File: src/Core/Services/Implementations/EventService.cs (hunk 1 above)
Comment (r-tome) [reply src/Core/Services/Implementations/EventService.cs:189]: That makes perfect sense, I'm happy with it being this way.

**Review by r-tome — APPROVED** [review]
Looks great to me, good job!

**Pull request merged**